zero-loads-app/src/loads/stub/sensor/definitions/units.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass
+from typing import Annotated, Any, Dict, TypeAlias, get_args, get_origin, get_type_hints
+
+from generator import Generator, GeneratorFactory
+from pydantic import (
+    AliasGenerator,
+    BaseModel,
+    ConfigDict,
+    Field,
+)
+
+from .util import hyphenize
+
+TOPIC_PREFIX: str = "sail-systems/"
+
+
+@dataclass(frozen=True)
+class Unit:
+    unit: str
+
+
+Position: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="mm")]
+RelativePosition: TypeAlias = Annotated[float, Field(ge=0, lt=1), Unit(unit="%")]
+Load: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="ton")]
+Torque: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="Nm")]
+RotationalSpeed: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="rpm")]
+Temperature: TypeAlias = Annotated[float, Field(ge=0), Unit(unit="°C")]
+
+
+class ComponentMeta(BaseModel):
+    topic: str
+
+
+def component_meta(*args, **kwargs):
+    return Field(json_schema_extra=ComponentMeta(*args, **kwargs).model_dump())
+
+
+class LoadsModel(BaseModel):
+    model_config = ConfigDict(
+        alias_generator=AliasGenerator(
+            serialization_alias=hyphenize,
+        )
+    )
+
+    @classmethod
+    def gen_config(cls, interval: int = 10) -> list[dict]:
+        """Generate configuration for data generation."""
+        hints = get_type_hints(cls, include_extras=True)
+        config = []
+
+        for component, value_definition in hints.items():
+            base_type, meta = cls._unwrap_annotated(value_definition)
+
+            topic = TOPIC_PREFIX + cls._extract_topic(meta, component)
+
+            if isinstance(base_type, type) and issubclass(base_type, LoadsModel):
+                values = {}
+                for field_name, field_info in base_type.model_fields.items():
+                    values[field_name] = cls._create_generator(field_info.annotation, field_info.metadata)
+
+                config.append(
+                    {
+                        "topic": topic,
+                        "interval": interval,
+                        "values": values,
+                    }
+                )
+            else:
+                config.append(
+                    {
+                        "topic": topic,
+                        "interval": interval,
+                        "values": {component: cls._create_generator(base_type, meta)},
+                    }
+                )
+
+        return config
+
+    @classmethod
+    def _extract_topic(cls, meta: list, component: str) -> str:
+        """Extract the topic from the metadata or use the component name as default."""
+        for m in meta:
+            if hasattr(m, "json_schema_extra"):
+                extra = getattr(m, "json_schema_extra", {})
+                if isinstance(extra, dict) and "topic" in extra:
+                    topic = extra["topic"]
+                    return topic
+
+        return component
```

Comment:
I'd just return `None` if no topic is found and `or component` over at the caller